src/main/java/com/softserve/demo/service/impl/EmailServiceImpl.java

```diff
@@ -42,42 +56,81 @@
+    private String getVerificationTemplate(final User user) {
+        String activationLink = HOSTLINK + "/register/verify/" + user.getActivationCode();
+        Context context = new Context();
+        context.setVariable("username", user.getUsername());
+        context.setVariable("activationLink", activationLink);
+        return templateEngine.process("verificationEmailTemplate", context);
+    }
+
+    /**
+     * This method creates an html template for the provided text.
+     *
+     * @param message text which will be wrapped with html
+     * @return html mail message
+     */
+    private String getMailTemplateWithText(final String message) {
+        Context context = new Context();
+        context.setVariable("message", message);
+        return templateEngine.process("emailTemplate", context);
+    }
+
+    /**
+     * Sends email with provided generated HTML template.
+     *
+     * @param addressedTo       address of the receiver
+     * @param subject           of the letter
+     * @param generatedTemplate template of the body of the letter
+     */
+    private void sendEmailWithTemplate(final String addressedTo, final String subject, final String generatedTemplate) {
         try {
             MimeMessage message = emailSender.createMimeMessage();
             MimeMessageHelper helper = new MimeMessageHelper(message, true);
             helper.setTo(addressedTo);
             helper.setSubject(subject);
-            helper.setText(getMailTemplateWithText(text), true);
+            helper.setText(generatedTemplate, true);
             emailSender.send(message);
         } catch (MessagingException e) {
             log.error("Unable to send email!", e);
-            return false;
         }
-        return true;
     }
 
     /**
-     * This method creates an html template for the provided text.
+     * This method sends email
+     * to the provided address.
      *
-     * @param message text which will be wrapped with html
-     * @return html mail message
+     * @param addressedTo receiver of the letter
+     * @param subject     subject of the letter
+     * @param text        the letter text itself
      */
-    private String getMailTemplateWithText(final String message) {
-        Context context = new Context();
-        context.setVariable("message", message);
-        return templateEngine.process("emailTemplate", context);
+    @Override
+    public void sendEmailTo(final String addressedTo, final String subject, final String text) {
+        singleThreadExecutor
+                .submit(() -> sendEmailWithTemplate(addressedTo, subject, getMailTemplateWithText(text)));
     }
+
+    /**
+     * This method sends Verification letter
+     * to the provided address.
+     *
+     * @param user from which all information is gathered
+     */
+    @Override
+    public void sendVerificationEmailTo(final User user) {
+        singleThreadExecutor
```

Comment:
Do we really need 2 methods?Why not to use the one with 3 params in both cases?

Reply:
I have written in this way because I am using two different templates. So I separate logic from creating templates. But we need different templates for verification and notification, that's why there is two methods. You have provided great review! And it is a big pleasure to read it and to get more experience! And I'm afraid that I need those two methods because of backward compatibility too :( someone has already used this service

Reply:
Well, logic for creation of templates is separated. But logic behind send email methods  is similar (it accepts 3 String params). As for backward compatibility, i believe,  this is not the case. You can simply use refactoring tools from IDEA to change method signature in all places it is used. Functionality will stay the same

Anyway, you can leave it as is and we can talk about this

Reply:
I think it will be great to discuss it face to face